packages/attestation-service/src/db.ts

```diff
@@ -53,38 +53,89 @@ export function isDBOnline() {
 }
 
 let kit: ContractKit | undefined
+let backupKit: ContractKit | undefined
+
+async function execWithFallback<T>(
+  f: (kit: ContractKit) => T,
+  kit1: ContractKit,
+  kit2?: ContractKit | undefined,
+  smartFallback = true
+): Promise<T> {
+  // Decorator to wrap execution of f with a retry using the kit2 kit
+  // if execution initially fails
+
+  let primaryKit = kit1
+  let secondaryKit = kit2
+  if (
+    smartFallback &&
+    kit2 &&
+    // TODO: look into whether we need to split this up out of the if statement
+    // check if not undefined, and have extra handling for that (i.e. take the other)
+
+    // Prioritize block age over syncing/not; avoids complex checks for stalls
+    // and assumes that if one is stalled, the next block number check will correct this.
+    (!kit1 ||
+      (await getAgeOfLatestBlockFromKit(kit2)).number >
+        (await getAgeOfLatestBlockFromKit(kit1)).number)
+  ) {
+    // Prioritize more-synced celo provider
+    rootLogger.info('Prioritizing CELO_PROVIDER_BACKUP')
+    primaryKit = kit2
+    secondaryKit = kit1
+  }
+  try {
+    return await f(primaryKit)
+  } catch (error) {
+    rootLogger.warn(`Using ContractKit failed: ${error}`)
+    if (secondaryKit) {
+      rootLogger.info(`Attempting to use secondary ContractKit`)
+      return await f(secondaryKit!)
+    } else {
+      throw error
+    }
+  }
+}
 
 // Wrapper that on error tries once to reinitialize connection to node.
 export async function useKit<T>(f: (kit: ContractKit) => T): Promise<T> {
-  if (!kit) {
+  const smartFallback = !(fetchEnvOrDefault('DISABLE_SMART_FALLBACK', 'false') === 'true')
```

Comment:
minor suggestion:
```suggestion
  const smartFallback = isYes(fetchEnvOrDefault('DISABLE_SMART_FALLBACK', 'false'))
```

Reply:
ooh yeah, I think I need to keep the `!` but changing this